tmt/steps/discover/__init__.py

```diff
@@ -415,6 +417,15 @@ def go(self, force: bool = False) -> None:
                                 test.serial_number == result.serial_number):
                             self._failed_tests[test_phase].append(test)
 
+        # Initialize results with pending outcome if no results for it available
+        results = self.plan.execute.results()
+        for index, (test, result) in enumerate(zip_longest(self.tests(), results)):
```

Comment:
I believe it won't be as easy as `zip()`. Results may be listed in an arbitrary order - yes, results are added once their tests finish, but I wouldn't rely on it, and there is certainly no code enforcing the order.

Check the `Execute.results_for_tests()` which lists results and test pairs, you should be interested in creating new results for pairs where the test is set and the result is `None`, that's a test that has not run yet.